src/sqlfluff/testing/rules.py

```diff
@@ -141,4 +141,12 @@ def rules__test_helper(test_case):
         )
         # If a `fixed` value is provided then check it matches
         if test_case.fix_str:
-            assert res == test_case.fix_str
+            assert res == test_case.fix_str, "result of fix doesn't match the fix_str"
+            assert_rule_pass_in_sql(
+                test_case.rule,
+                test_case.fix_str,
+                configs=test_case.configs,
+            )
+        else:
+            # check that tests without a fix_str do no apply ANY fixes
+            assert res == test_case.fail_str, "fix should not have modified the query"
```

Comment:
This assertion is straightforward. If a test fails on this assertion, it can be fixed by adding the `fix_str`. I could make a new PR first that only adds this assertion?